server/routerlicious/packages/lambdas/src/deli/lambdaFactory.ts

```diff
@@ -225,6 +227,38 @@ export class DeliLambdaFactory
 					closeType === LambdaCloseType.ActivityTimeout ||
 					closeType === LambdaCloseType.Error
 				) {
+					if (document?.isEphemeralContainer) {
+						// Call to historian to delete summaries
+						await requestWithRetry(
+							async () => gitManager.deleteSummary(false),
+							"deliLambda_onClose" /* callName */,
+							getLumberBaseProperties(documentId, tenantId) /* telemetryProperties */,
+							(error) => true /* shouldRetry */,
+							3 /* maxRetries */,
+						);
+
+						// Delete the document metadata
+						await runWithRetry(
+							async () =>
+								this.documentRepository.deleteOne({
+									documentId,
+									tenantId,
+								}),
+							"deleteDocMetadata",
+							3 /* maxRetries */,
```

Comment:
What if this fails, but the call from line 232 above succeeds? Would it be a problem to still have the document info in the DB, but no more summary data in (redis) storage?

Reply:
Perhaps we can add a try/catch here with specific error handling and logging for those situations? We do have try/catch in runWithRetry and requestWithRetry but those are more generic.

Reply:
That would not be a problem. The only difference is that it will fail at a different point where it will not be able to fetch the summary. It is to handle this scenario that I had to return a 404 instead of a 500 for ephemeral containers, so that the clients can react accordingly.

I think these calls are not important enough to track individually. Since the session is already winding down, these failing should not affect the session. It is just that the session cleanup will happen at a later point when the TTL expires. Also, the error stack for both the calls are quite unique, so it should be sufficient to tell them apart.

Reply:
Sounds good. I agree we should be good since all the data will have a TTL.